File: src/0220-Contains-Duplicate-III/0220.cpp

#include <iostream>
#include <vector>
#include <unordered_map>
#include <set>
using namespace std;

static int x = []() {std::ios::sync_with_stdio(false); cin.tie(0); return 0; }();
class Solution
{
public:
    bool containsNearbyAlmostDuplicate(vector<int>& nums, int k, int t) 
    {
        set<long long> record;
        for (int i = 0; i < nums.size(); ++i)
        {
            if (record.lower_bound((long long)nums[i] - (long long)t) != record.end() && 
                    *record.lower_bound((long long)nums[i] - (long long)t) <= (long long)nums[i] + (long long)t)
                return true;
            record.insert(nums[i]);
            if (record.size() == k + 1)
            {
                record.erase(nums[i - k]);
            } 
        }
        return false;
    }
};
int main()
{
    vector<int> nums = {1,2,3,1};
    int k = 4;
    int t = 0;
    cout << Solution().containsNearbyAlmostDuplicate(nums, k, t);
    return 0;
}

File: src/0220-Contains-Duplicate-III/0220.py
class Solution:
    def containsNearbyAlmostDuplicate(self, nums, k, t):
        """
        :type nums: List[int]
        :type k: int
        :type t: int
        :rtype: bool
        """
        for i, a in enumerate(nums):
            for j, b in enumerate(nums):
                if i != j and abs(i - j) <= k and abs(a - b) <= t:
                    return True

        return False
        

if __name__ == "__main__":
    nums = [4,2]
    k = 2
    t = 1
    print(Solution().containsNearbyAlmostDuplicate(nums, k, t))